### PR TITLE
[improve][broker]prevent new connection request when broker is closing

### PR DESCRIPTION
**Pull request by HQebupt**
### Motivation
Inspired by #17085 
When the leader broker is closing, and the client will reconnect and redo lookup, resulting in unexpected results. 

> @codelipenghui 
> Looks like we also need to check other cases (e.g. handleConnect)
> If the broker is shutting down, we should also prevent new connections.

Therefore, it is necessary to check the status of the broker when handling requests, including connect\lookup\partitioned metadata.

The following is the interaction workflow between consumer and broker.
```
Consumer -> Broker: CONNECT
Broker -> Consumer: CONNECTED
Consumer -> Broker: PARTITIONED_METADATA
Broker -> Consumer: PARTITIONED_METADATA_RESPONSE
Consumer -> Broker: LOOKUP
Broker -> Consumer: LOOKUP_RESPONSE
Consumer -> Broker: SUBSCIBE
Broker -> Consumer: SUCCESS
Consumer -> Broker: FLOW
Broker -> Consumer: MESSAGE
Consumer - Broker: ACK
```

The following screenshot is the interaction workflow between producer and broker.
image

### Modifications
- Check Pulsar Service state when handle connection and partitioned metadata request.

### Verifying this change

- [x]  Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.
### Does this pull request potentially affect one of the following parts:

If `yes` was chosen, please highlight the changes

- Dependencies (does it add or upgrade a dependency): (no)
- The public API: (no)
- The schema: (no)
- The default values of configurations: (no)
- The wire protocol: (no)
- The rest endpoints: (no)
- The admin cli options: (no)
- Anything that affects deployment: (no)

### Documentation
Check the box below and label this PR (if you have committer privilege).

Need to update docs? 
- [x] `doc-not-needed`

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java [hunk 1]
```diff
@@ -817,6 +817,17 @@ protected void handleConnect(CommandConnect connect) {
             return;
         }
 
+        if (!this.service.getPulsar().isRunning()) {
```
Comment (AnonHxy) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java:820]: If `!service.isAuthenticationEnabled()` return true, the connect will complete without checking pulsar service  is running or not.

 My suggestion is that move this code block just to Line807,  which is just below the debug log @HQebupt

**Review by codelipenghui — APPROVED** [review]

**Comment by codelipenghui** [comment]
Please add a unit test to cover the new change

**Comment by HQebupt** [comment]
> Please add a unit test to cover the new change

done

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java (hunk 1 above)
Comment (HQebupt) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java]: Resolved, thank you. PTAL @AnonHxy @congbobo184

**Review by Technoboy- — APPROVED** [review]

**Review by michaeljmarshall — APPROVED** [review]
I understand the need for this change and the one in https://github.com/apache/pulsar/pull/17085, but I wonder if it means that our start up and shut down logic needs to be revisited so that all of this logic doesn't get added to the `ServerCnx` class.

With that said, this patch will fix a specific bug, so I think we should merge it. LGTM.

**Review by AnonHxy — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java (hunk 1 above)
Comment (AnonHxy) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java]: LGTM now

**Pull request merged**